internal/pkg/manifest/svc.go

```diff
@@ -495,6 +495,46 @@ func (rr RoutingRuleConfiguration) exposedPorts(exposedPorts []ExposedPort, work
 	}
 }
 
+// exportPorts returns any new ports that should be exposed given the network load balancer
+// configuration that's not part of the existing containerPorts.
+func (cfg NetworkLoadBalancerConfiguration) exposedPorts(exposedPorts []ExposedPort, workloadName string) ([]ExposedPort, error) {
+	if cfg.IsEmpty() {
+		return nil, nil
+	}
+	containerName := workloadName
+	if cfg.TargetContainer != nil {
+		containerName = aws.StringValue(cfg.TargetContainer)
+	}
+	nlbPort, protocol, err := ParsePortMapping(cfg.Port)
+	if err != nil {
+		return nil, err
+	}
+	if protocol == nil {
+		protocol = aws.String("tcp")
+	}
+
+	port, err := strconv.ParseUint(aws.StringValue(nlbPort), 10, 16)
+	if err != nil {
+		return nil, err
+	}
+	containerPort := uint16(port)
+	if cfg.TargetPort != nil {
+		containerPort = uint16(aws.IntValue(cfg.TargetPort))
+	}
+	for _, exposedPort := range exposedPorts {
+		if containerPort == exposedPort.Port && cfg.TargetContainer == nil {
```

Comment:
what if they set `TargetContainer` to be the main container? This condition would not be satisfied and we'll end up exposing a duplicated port for the main container 🤔

Reply:
I think removing `cfg.TargetContainer == nil` resolves all the problems. I thought I committed the change of removing this `cfg.TargetContainer == nil` condition before going on a vacation. I will commit it asap.